flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServer.java

```diff
@@ -174,7 +176,6 @@ public ServerSocket createSocket(int port) throws IOException {
 		// start the server thread
 		setName("BLOB Server listener at " + getPort());
 		setDaemon(true);
-		start();
```

Comment:
Why did you pull `start` out of the constructor? Wouldn't one always want to start the `BlobServer` when creating it?

Reply:
Basically yes but this pattern does not work well with classes inheriting from `BlobServer`. Here, this is only the case for the `TestingFailingBlobServer` which sets additional properties and relies on them being set while executing `run()`. If started in the constructor, the `run()` method would work on a partially initialised object.

Reply:
Makes sense.